multi-int-store/src/main/java/com/connexta/multiintstore/services/impl/IndexedMetadataDao.java

```diff
@@ -23,16 +25,34 @@ public IndexedMetadataDao(IndexedMetadataRepository repository) {
 
   @Override
   public Optional<IndexedProductMetadata> getById(String id) {
-    return repository.findById(id);
+    try {
+      return repository.findById(id);
+    } catch (RuntimeException e) {
+      throw new StorageException("Could not connect to Solr");
+    }
   }
 
   @Override
   public void save(IndexedProductMetadata indexedProductMetadata) {
-    repository.save(indexedProductMetadata);
+    Optional<IndexedProductMetadata> document = getById(indexedProductMetadata.getId());
+    document.ifPresent(
+        (metadata) -> {
+          throw new DuplicateIdException(
+              "Metadata with id " + metadata.getId() + " already exists");
+        });
+    try {
+      repository.save(indexedProductMetadata);
+    } catch (RuntimeException e) {
+      throw new StorageException("Could not connect to Solr");
```

Comment:
❓ Could something other than not being able to connect cause a RuntimeException?

Reply:
Maybe, I'm unsure though. I have a hunch for our situation if this code is running it's due to a connection error. I'll make it more general